requirements.txt

```diff
@@ -1,5 +1,2 @@
-git+https://github.com/pyinstaller/pyinstaller.git
-Click
-python-docx
-cairosvg
-pygal
+git+https://github.com/pyinstaller/pyinstaller.git#egg=pyinstaller
+-e file:.[dev]
```

Comment:
Hi @altendky !
I am not familiar with this second line, could you explain what it does?

Reply:
It is for installing allure-docx itself.  Generically, it's for installing the directory containing the `requirements.txt` file (`file:.`, often written just `.`) in editable mode (`-e`, I should change that to `--editable`) including the development extras (`[dev]` which refers to the `'dev'` key in `setup.py`'s `extras_require` dict).

Mostly this file wasn't doing anything `setup.py` itself can't except for the PyInstaller Git URL.  Well, it can even do that via `dependency_links` but that's a frowned upon feature afaik.  So, I moved all the deps into `setup.py`.  In general I would expect a `requirements.txt` to additionally specify specific versions of libraries which are to be used for a recreatable environment to develop and test in.  Normally also you wouldn't actually have the `-e .` in a `requirements.txt`, and yes, I know I just stuck it there...

Anyways, as the PR sits now you could create a virtualenv/venv and `yourenv/bin/pip install -r requirements.txt` and get allure-docx installed editable along with all the deps including the Git-sourced PyInstaller.  Depending how you actually want to use this file we could adjust it to meet that need.  If you aren't developing and are just using the project then a regular old `yourenv/bin/pip install .` will get allure-docx installed with all the deps (and without touching this `requirements.txt`).

Reply:
Thanks for the explanation, looks great!